match2/commons/match_group_display_matchlist.lua

```diff
@@ -1,217 +1,291 @@
-	local tbd2 =
-		match.opponent2template == 'tbd' or match.opponent2 == 'TBD' or
-		utils.string.startsWith(opponent2data.type, 'literal')
-
-	local hasDetails = HasDetails(match)
-
-	return html.create("tr")
-		:addClass("brtks-matchlist-row brkts-match-popup-wrapper")
-		:css("cursor", "pointer")
-		:node(html.create("td")
-			:addClass("brkts-matchlist-slot brkts-opponent-hover" .. (tonumber(match.winner) == 1 and " brkts-matchlist-slot-winner" or tonumber(match.winner) == 0 and " brkts-matchlist-slot-bold bg-draw" or ""))
-			:attr("aria-label", not tbd1 and opponent1key or nil)
-			:attr("width", "40%")
-			:attr("align", "right")
-			:node(OpponentDisplay.luaGet(_frame, _addDisplayType(_flattenArgs(opponent1data), "matchlist-left")))
-		)
-		:node(html.create("td")
-			:addClass("brkts-matchlist-slot brkts-opponent-hover" .. ((tonumber(match.winner) == 1 or tonumber(match.winner) == 0) and " brkts-matchlist-slot-bold" or ""))
-			:attr("aria-label", not tbd1 and opponent1key or nil)
-			:attr("width", "10.8%")
-			:attr("align", "center")
-			:node(OpponentDisplay.luaGet(_frame, _addDisplayType(_flattenArgs(opponent1data), "matchlist-left-score")))
-		)
-		:node(
-			not hasDetails and "" or
-			html.create("td")
-				:addClass("brkts-match-info brkts-empty-td")
-				:node(
-					html.create("div")
-						:addClass("brkts-match-info-icon")
-				)
-				:node(
-					html.create("div")
-						:addClass("brkts-match-info-popup")
-						:node(MatchSummary.luaGet(_frame, _flattenArgs(match)))
-						:css("display", "none")
-				)
-			)
-		:node(html.create("td")
-			:addClass("brkts-matchlist-slot brkts-opponent-hover" .. ((tonumber(match.winner) == 2 or tonumber(match.winner) == 0) and " brkts-matchlist-slot-bold" or ""))
-			:attr("aria-label", not tbd2 and opponent2key or nil)
-			:attr("width", "10.8%")
-			:attr("align", "center")
-			:node(OpponentDisplay.luaGet(_frame, _addDisplayType(_flattenArgs(opponent2data), "matchlist-right-score")))
-		)
-		:node(html.create("td")
-			:addClass("brkts-matchlist-slot brkts-opponent-hover" .. (tonumber(match.winner) == 2 and " brkts-matchlist-slot-winner" or tonumber(match.winner) == 0 and " brkts-matchlist-slot-bold bg-draw" or ""))
-			:attr("aria-label", not tbd2 and opponent2key or nil)
-			:attr("width", "40%")
-			:attr("align", "left")
-			:node(OpponentDisplay.luaGet(_frame, _addDisplayType(_flattenArgs(opponent2data), "matchlist-right")))
-		)
+--[[
+Display component for a match in a matchlist. Consists of two opponents, two 
+scores, and a icon for the match summary popup.
+]]
+Matchlist.propTypes.Match = {
+    MatchSummaryContainer = 'function',
+    Opponent = 'function',
+    Score = 'function',
+    match = MatchGroupUtil.types.Match,
+    matchHasDetails = 'function',
+}
+function Matchlist.Match(props)
+    DisplayUtil.assertPropTypes(props, Matchlist.propTypes.Match)
+    local match = props.match
+    
+    local renderOpponent = function(opponentIx)
```

Comment:
Might be better to just make these proper functions with how much space they take up?

Reply:
I'll be rewriting this when converting OpponentDisplay to use structural types for match2opponent.